### ISSUE TITLE
Windows Universal - Some issues

### ISSUE THREAD

**Issue by SunboX** [Redth/ZXing.Net.Mobile#223]
First, let me say ... thank you, thank you, thank you! :smile: :heart: 

I was waiting for this too long ...

I tried your current Windows Universal sample (master of 2015-12-18) on a Lumia 950.

The app starts, but
* Scanning QR-Code shows a message `"Sometimes Being simple is much better. Be simple!"`
* Scanning EAN13 codes seems not to work at all
* Pressing the back button does not cancel the scanner overlay, but does close the whole app

I know, this is still in the work. Just wanted to say thanks and give some feedback! :smirk: If you need someone to test these things, let me know (we have some different mobiles here).

## Output
```
'Sample.WindowsUniversal.exe' (CoreCLR: DefaultDomain): Loaded 'C:\data\Programs\WindowsApps\Microsoft.NET.CoreRuntime.1.0_1.0.23430.0_arm__8wekyb3d8bbwe\mscorlib.ni.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\entrypoint\Sample.WindowsUniversal.exe'. Symbols loaded.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\System.Runtime.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\WinMetadata\Windows.winmd'. Module was built without symbols.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\System.Runtime.InteropServices.WindowsRuntime.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\Microsoft.ApplicationInsights.Extensibility.Windows.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\System.Threading.Tasks.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\Microsoft.ApplicationInsights.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\System.Runtime.WindowsRuntime.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\System.Diagnostics.Debug.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\System.Runtime.WindowsRuntime.UI.Xaml.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\System.Collections.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\ZXingNetMobile.dll'. Symbols loaded.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\ZXing.Net.Mobile.Core.dll'. Symbols loaded.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\zxing.portable.dll'. Symbols loaded.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\System.Private.Uri.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\Microsoft.ApplicationInsights.PersistenceChannel.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\System.Threading.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\System.Diagnostics.Tracing.dll'. Module was built without symbols.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\System.Linq.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\System.Globalization.dll'. Module was built without symbols.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\System.IO.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\System.Runtime.Extensions.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\System.Reflection.dll'. Module was built without symbols.
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\System.Reflection.Extensions.dll'. Module was built without symbols.
The thread 0x3e4 has exited with code 0 (0x0).
The thread 0x6f8 has exited with code 0 (0x0).
The thread 0x115c has exited with code 0 (0x0).
The thread 0x1010 has exited with code 0 (0x0).
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\System.Threading.Timer.dll'. Module was built without symbols.
Camera Preview Resolution: 1920x1080
Camera Preview Resolution: 1280x720
Camera Preview Resolution: 800x450
Camera Preview Resolution: 1440x1080
Camera Preview Resolution: 1024x768
Camera Preview Resolution: 640x480
Using Preview Resolution: 800x450
Exception thrown: 'System.IndexOutOfRangeException' in zxing.portable.dll
Exception thrown: 'System.IndexOutOfRangeException' in zxing.portable.dll
'Sample.WindowsUniversal.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\SharedData\PhoneTools\AppxLayouts\80a2fbc7-843e-46ca-a740-cbb1bc604d33VS.Debug_ARM.afiedler\System.Text.Encoding.dll'. Module was built without symbols.
Exception thrown: 'System.Runtime.InteropServices.COMException' in mscorlib.ni.dll
WinRT information: The op is invalid.
The thread 0x478 has exited with code 0 (0x0).
The thread 0xbc0 has exited with code 0 (0x0).
The thread 0x1420 has exited with code 0 (0x0).
Camera Preview Resolution: 1920x1080
Camera Preview Resolution: 1280x720
Camera Preview Resolution: 800x450
Camera Preview Resolution: 1440x1080
Camera Preview Resolution: 1024x768
Camera Preview Resolution: 640x480
Using Preview Resolution: 800x450
Exception thrown: 'System.Runtime.InteropServices.COMException' in mscorlib.ni.dll
WinRT information: The op is invalid.
The thread 0x4dc has exited with code 0 (0x0).
The thread 0xff0 has exited with code 0 (0x0).
Camera Preview Resolution: 1920x1080
Camera Preview Resolution: 1280x720
Camera Preview Resolution: 800x450
Camera Preview Resolution: 1440x1080
Camera Preview Resolution: 1024x768
Camera Preview Resolution: 640x480
Using Preview Resolution: 800x450
AutoFocus requested
AutoFocus requested
The thread 0x6f8 has exited with code 0 (0x0).
The thread 0x11b4 has exited with code 0 (0x0).
Exception thrown: 'System.IndexOutOfRangeException' in zxing.portable.dll
Exception thrown: 'System.Runtime.InteropServices.COMException' in mscorlib.ni.dll
WinRT information: The op is invalid.
The thread 0x115c has exited with code 0 (0x0).
The thread 0xadc has exited with code 0 (0x0).
The thread 0xbc has exited with code 0 (0x0).
Camera Preview Resolution: 1920x1080
Camera Preview Resolution: 1280x720
Camera Preview Resolution: 800x450
Camera Preview Resolution: 1440x1080
Camera Preview Resolution: 1024x768
Camera Preview Resolution: 640x480
Using Preview Resolution: 800x450
The thread 0x78c has exited with code 0 (0x0).
AutoFocus requested
Exception thrown: 'System.Exception' in mscorlib.ni.dll
Exception thrown: 'System.Runtime.InteropServices.COMException' in mscorlib.ni.dll
WinRT information: The op is invalid.
The thread 0x105c has exited with code 0 (0x0).
The thread 0x1150 has exited with code 0 (0x0).
The thread 0x404 has exited with code 0 (0x0).
Camera Preview Resolution: 1920x1080
Camera Preview Resolution: 1280x720
Camera Preview Resolution: 800x450
Camera Preview Resolution: 1440x1080
Camera Preview Resolution: 1024x768
Camera Preview Resolution: 640x480
Using Preview Resolution: 800x450
Exception thrown: 'System.IndexOutOfRangeException' in zxing.portable.dll
Exception thrown: 'System.IndexOutOfRangeException' in zxing.portable.dll
AutoFocus requested
Exception thrown: 'System.IndexOutOfRangeException' in zxing.portable.dll
AutoFocus requested
AutoFocus requested
AutoFocus requested
AutoFocus requested
The thread 0x1170 has exited with code 0 (0x0).
AutoFocus requested
The thread 0xe60 has exited with code 0 (0x0).
AutoFocus requested
The program '[5900] Sample.WindowsUniversal.exe' has exited with code -1 (0xffffffff).
```

**Comment by Redth** [comment]
So,

1. This was a simple oversight in the sample and is fixed now.
2. It looks like the preview buffer is rotated... I need to change code to account for this.  For now you can either rotate your device, or set `TryHarder` to true in the options.
3. This may be me not knowing how to properly handle back navigation...

**Comment by Redth** [comment]
So I think #224 should fix #3.
Closing this and will track the remaining issue in #241

**Comment by SunboX** [comment]
n.p. ... thanks for your work! :+1:
